### DISCUSSION TITLE
Stripe plugin fails to include shipping in price

### DISCUSSION THREAD

**Discussion by agoransson** [vendurehq/vendure#2438] [Q&A]
Hi, I've been trying to get the stripe plugin behaving as expected - but there's something I'm missing I think because at every single order I place the shipping price is left out, and the order never progresses from "Arranging payment".

According to code it should use the active order for the request context, however I get this error in the log:

`[Vendure Server] USER_INPUT_ERROR: No active order found for session`

Which I located to this code:

```
    async createStripePaymentIntent(@Ctx() ctx: RequestContext): Promise<string> {
        if (!ctx.authorizedAsOwnerOnly) {
            throw new UnauthorizedError();
        }
        const sessionOrder = await this.activeOrderService.getActiveOrder(ctx, undefined);
        if (!sessionOrder) {
            throw new UserInputError('No active order found for session');
        }
        return this.stripeService.createPaymentIntent(ctx, sessionOrder);
    }
```

**Comment by michaelbromley** [comment]
Hi,

The error suggests some kind of issue in communicating the session token to the server when calling `createStripePaymentIntent`. This could be an issue with cookies (if you are using cookie-based sessions) or with the bearer token not being sent correctly (if you are using header-based sessions).

**Reply by agoransson** [reply]
For some reason I've been required to use auth tokens since some time now, it's not been enough to just include the cookies. I did check the mutation request and the token is included as expected in the headers. I'm using ApolloClient, and my setup looks almost identical to that of the example given. 

```
const client = new ApolloClient({
    link: ApolloLink.from([
        setContext(() => {
            const authToken = localStorage.getItem(AUTH_TOKEN_KEY)
            if (authToken) { 
                return {
                    headers: {
                        authorization: `Bearer ${authToken}`,
                    },
                }
            }
        }),
        afterwareLink,
        httpLink,
    ]),
    cache: new InMemoryCache(CacheConfig),
});
```

Request headers from my createStripePaymentIntent mutation:
```
:authority:
www.my-vendure-server-address.com
:method:
POST
:path:
/shop-api?languageCode=
:scheme:
https
Accept:
*/*
Accept-Encoding:
gzip, deflate, br
Accept-Language:
sv-SE,sv;q=0.9
Authorization:
Bearer fe446ff990616....179eda6967ffaa7
Content-Length:
139
Content-Type:
application/json
Dnt:
1
Origin:
http://localhost:4000
Referer:
http://localhost:4000/
Sec-Ch-Ua:
"Chromium";v="118", "Google Chrome";v="118", "Not=A?Brand";v="99"
Sec-Ch-Ua-Mobile:
?0
Sec-Ch-Ua-Platform:
"macOS"
Sec-Fetch-Dest:
empty
Sec-Fetch-Mode:
cors
Sec-Fetch-Site:
cross-site
User-Agent:
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36
```

Stripe seems to be happy, so I get a response back.

```
{
    "data": {
        "createStripePaymentIntent": "pi_3O31g..."
    }
}
```

Could it be a problem that the vendure server is running in the cloud, while my dev store is running locally? I don't see how this can be a problem...

**Reply by agoransson** [reply]
It appears that I created the payment intent just one step too soon, even if the shipping had been selected in the system (as viewed in admin page) the order for some reason didn't include the shipping price. When I moved the creation of the payment intent "one step closer" to the actual card details it would work.
